src/articles/SubscribeSearchButton.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from "react";
+import * as s from "./SubscribeSearchButton.sc";
+import useSelectInterest from "../hooks/useSelectInterest";
+import { toast } from "react-toastify";
+import SubscribeToEmailNotifications from "./SubscribeToEmailNotifications";
+
+export default function SubscribeSearchButton({ api, query }) {
+  const { subscribedSearches, removeSearch, subscribeToSearch } =
+    useSelectInterest(api);
+
+  const [textButton, setTextButton] = useState("");
+  const [isSubscribedToSearch, setIsSubscribedToSearch] = useState(false);
+
+  useEffect(() => {
+    if (subscribedSearches) {
+      const isSubscribed = subscribedSearches.some(
+        (search) => search.search === query,
+      );
+      setIsSubscribedToSearch(isSubscribed);
+      setTextButton(isSubscribed ? "- remove search" : "+ add search");
+    }
+  }, [subscribedSearches, query]);
+
+  const toggleSearchSubscription = (query) => {
+    if (isSubscribedToSearch) {
+      const searchToRemove = subscribedSearches.find(
+        (search) => search.search === query,
+      );
+      if (searchToRemove) {
+        removeSearch(searchToRemove);
+        setIsSubscribedToSearch(false);
+        toast("Search removed from My Searches!");
+      }
+    } else {
+      subscribeToSearch(query);
+      setIsSubscribedToSearch(true);
+      toast("Search added to My Searches!");
+    }
+  };
+
+  return (
+    <>
+      <s.AddRemoveButton onClick={(e) => toggleSearchSubscription(query)}>
```

Comment:
query does not need to be passed as argument here. 
it's available in the context of the function as the prop at the top.
thus, this can be written simpler as:

```js
<s.AddRemoveButton onClick={toggleSearchSubscription}>
```

and the implementation of `toggleSearchSubscription` above can also drop the `query` parameter.